src/middleware/rolePermission.js

```diff
@@ -0,0 +1,20 @@
+const roles = require('../constants/roles.js')
+const { auth } = require('../constants/errors.json')
+const boom = require('boom')
+
+// Recieves req.user from addUserIdToSession, and we extract the user's role from that
+const rolePermissionIsSufficient = ({ minRole }) => user => {
+  const rolesOrder = [roles.BASIC, roles.ADMIN, roles.SUPER]
+
+  return (rolesOrder.indexOf(user.role) >= rolesOrder.indexOf(minRole))
```

Comment:
Yes, this is so much nicer!

Reply:
One problem that I see is that `-1 === -1`. This is not a big issue but you could

```js
const rolePermissionIsSufficient = ({ minRole }) => {
  const rolesOrder = [roles.BASIC, roles.ADMIN, roles.SUPER]
  if (rolesOrder.indexOf(minRole) === -1) {
    throw new Error(..) // this will throw during startup so no problem
  }
  return user => {
    ...
  }
}
```